client/src/js/components/bhPatientFinancialActivity.js

```diff
@@ -71,8 +71,6 @@ function PatientFinancialActivityCtrl(Patients, moment, Session, Constants) {
 
         $ctrl.dataMovement = dataMovement;
         $ctrl.noStockMovement = stockData.length === 0;
-      })
-      .finally(() => {
```

Comment:
Aah, I see you noticed the loading bug.  The actual problem isn't here.   If you want to fix it, change the `Promise.all()` at the beginning of this chain  to `$q.all()`.

AngularJS doesn't run the $digest loop after the `Promise()` but does after `$q()`.